gap/semigroups/semitrans.gi

```diff
@@ -283,49 +283,72 @@ SEMIGROUPS.LargestElementRClass := function(R)
   return SEMIGROUPS.ElementRClass(R, true);
 end;
 
-# stop_on_isolated_pair:
-#   if true, this function returns false if there is an isolated pair-vertex
-# TODO should this be a Digraph??
-# FIXME make this a digraph
-SEMIGROUPS.GraphOfRightActionOnPairs := function(gens, n, stop_on_isolated_pair)
-  local nrgens, nrpairs, PairNumber, NumberPair, in_nbs, labels, pair,
-  isolated_pair, act, range, i, j;
-  nrgens     := Length(gens);
-  nrpairs    := Binomial(n, 2);
+InstallMethod(DigraphOfActionOnPairs,
+"for a transformation semigroup",
+[IsTransformationSemigroup],
+function(S)
+  return DigraphOfActionOnPairs(GeneratorsOfSemigroup(S),
+                                DegreeOfTransformationSemigroup(S));
```

Comment:
Isn't the `GeneratorsOfSemigroup(S)` here unnecessary? Since you anyway check in the main function if a collection is a semigroup and if it is replace it with the generators? Doesn't that make this whole method unnecessary?